bc/core/utils/status/templates.py

```diff
@@ -50,23 +45,30 @@
 )
 
 MASTODON_FOLLOW_A_NEW_CASE = MastodonTemplate(
-    link_placeholders=[],
-    str_template="""I'm now following {docket}:
-
-{docket_link}
-
-#CL{docket_id}""",
-)
-
-MASTODON_FOLLOW_A_NEW_CASE_W_ARTICLE = MastodonTemplate(
-    link_placeholders=["article_url"],
-    str_template="""I'm now following {docket}:
-
-Docket: {docket_link}
-
-Context: {article_url}
-
-#CL{docket_id}""",
+    link_placeholders=["docket_link", "initial_complaint_link", "article_url"],
+    # Remove extra newlines caused by empty template blocks
+    post_replace=[
+        ("\n\n\n\n", "\n\n"),
+        ("\n\n\n\n", "\n\n"),
+        ("\n\n\n", "\n\n"),
+    ],
+    str_template="""I'm now following {{docket}}:
+
+{% if date_filed %}
+Filed: {{date_filed}}
+{% endif %}
+
+Docket: {{docket_link}}
+
+{% if initial_complaint_type and initial_complaint_link %}
+{{initial_complaint_type}}: {{initial_complaint_link}}
+{% endif %}
+
+{% if article_url %}
+Context: {{article_url}}
+{% endif %}
+
+#CL{{docket_id}}""",
 )
```

Comment:
We deal with this in plain text emails too. I think a more elegant (?) solution is to put your newlines inside your if statements. I'm going to get this wrong, but one approach is something like: 

```suggestion
    str_template="""I'm now following {{docket}}:{% if date_filed %}
Filed: {{date_filed}}{% endif %}
Docket: {{docket_link}}{% if initial_complaint_type and initial_complaint_link %}
{{initial_complaint_type}}: {{initial_complaint_link}}{% endif %}{% if article_url %}
Context: {{article_url}}{% endif %}

#CL{{docket_id}}""",
)
```

Reply:
Completely agree. That felt like the ugliest bit. I had tried a few ways to avoid it, but never thought of putting the `if`s on the preceding lines!

I have it working now without the `post_replace`. Thanks!

Reply:
Yeah, it's still not pretty, per se, but at least it's self-documenting and hopefully not as fiddly. Thanks for the refactor.